### ISSUE TITLE
UBL issue w/ BLTouch

### ISSUE THREAD

**Issue by modem7** [MarlinFirmware/Marlin#10963]
### Description

Note: This was working as expected in 1.1.4

Marlin 1.1.8
https://github.com/MarlinFirmware/Marlin/archive/1.1.x.zip

When doing step by step UBL, getting an issue where the points aren't being printed across the whole bed (inset set to 10mm), and once it gets to a certain point, the z moves up to infinity and crashes into the frame.

### Steps to Reproduce

1. Step by Step UBL
2. Verify mesh PLA

[2018-05-31 17-27-37.zip](https://github.com/MarlinFirmware/Marlin/files/2085437/2018-05-31.17-27-37.zip)
![2018-06-08 18 33 56](https://user-images.githubusercontent.com/4349962/41172150-d9e76c82-6b4a-11e8-8edc-a2cca71acf20.jpg)

**Comment by gloomyandy** [comment]
What version of Marlin are you using? You should probably try a recent bugfix version as many fixes and changes have been made to UBL in recent months.

**Comment by modem7** [comment]
Hey @gloomyandy - 1.1.8, whichever is the latest version

https://github.com/MarlinFirmware/Marlin/archive/1.1.x.zip

**Comment by gloomyandy** [comment]
Have you tried either bugfix-1.1.x or bugfix-2.0.x? If not then you really should, many changes in these versions and you are unlikely to get anything fixed in 1.1.8.

**Comment by modem7** [comment]
Will check that now!

**Comment by modem7** [comment]
OK, with the latest bugfix version (https://github.com/MarlinFirmware/Marlin/archive/bugfix-1.1.x.zip) I get the head trying to crash into the bed in infinity (opposite direction as before, so at least some progress?), and it's still not doing the mesh test across the entire bed. 

Almost seems to be related to #6581, but not quite sure, I'm at a loss.
```
21:40:33.212 : Bed Topography Report:
21:40:33.243 : (0,9)                                                                   (9,9)
21:40:33.259 : (34,150)                                                                (190,150)
21:40:33.493 : 0.543  [ 0.287]   0.170    0.018   -0.114   -0.314   -0.475   -0.627   -0.929
21:40:33.493 : 
21:40:33.664 : 0.493    0.406    0.214    0.152   -0.010   -0.179   -0.244   -0.446   -0.767   -1.013
21:40:33.664 : 
21:40:33.820 : 0.548    0.279    0.232    0.140   -0.010   -0.094   -0.321   -0.461   -0.655   -0.944
21:40:33.820 : 
21:40:33.976 : 0.498    0.309    0.224    0.080    0.003   -0.206   -0.246   -0.685   -0.829   -1.028
21:40:33.976 : 
21:40:34.132 : 0.543    0.274    0.182    0.062    0.048   -0.254   -0.299   -0.533   -0.772   -0.976
21:40:34.132 : 
21:40:34.288 : 
21:40:34.288 : 
21:40:34.444 : 0.481    0.456    0.262    0.085    0.010   -0.214   -0.423   -0.503   -0.700   -0.976
21:40:34.444 : 
21:40:34.600 : 0.628    0.386    0.356    0.145    0.023   -0.199   -0.411   -0.523   -0.819   -0.971
21:40:34.600 : 
21:40:34.756 : 0.580    0.416    0.351    0.085    0.030   -0.102   -0.461   -0.515   -0.819   -1.013
21:40:34.756 : 
21:40:34.912 : 0.580    0.369    0.219    0.155   -0.030   -0.229   -0.461   -0.680   -0.792   -0.988
21:40:34.943 : (34,10)                                                                    (190,10)
21:40:34.959 : (0,0)                                                                     (9,0)
21:40:37.455 : echo:enqueueing "G28"
```

[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/2086051/Marlin.zip)

![2018-06-08 21 58 14](https://user-images.githubusercontent.com/4349962/41180875-a83a4c3c-6b67-11e8-94c6-4258fa9e926a.jpg)

**Comment by thinkyhead** [comment]
You may also want to test with `AUTO_BED_LEVELING_BILINEAR` so we can determine whether there's a UBL-specific issue or a more general issue with `G26`.

Note that with `AUTO_BED_LEVELING_UBL` points outside of the mesh are simply _not handled_, so the nozzle will raise up to the height set in `UBL_Z_RAISE_WHEN_OFF_MESH` for anyplace outside the mesh. Thus it's important with UBL to make the mesh itself cover the whole bed (`#define MESH_INSET 0`), allow UBL to probe the points that _can_ be reached by the probe first with `G29 P1`, then use `G29 P2` to fill in the points that _cannot_ be reached by the probe, and then save the mesh to EEPROM.

**Comment by modem7** [comment]
I'll give the mesh_inset 0 a try tomorrow, and then work my way back!

Appreciate the heads up!

**Comment by Roxy-3D** [comment]
Yes!  Just walk through the 'Cheat Sheet' here:  https://github.com/MarlinFirmware/MarlinDocumentation/blob/master/_features/unified_bed_leveling.md
```
;------------------------------------------
;--- Setup and initial probing commands ---
;------------------------------------------
M502            ; Reset settings to configuration defaults...
M500            ; ...and Save to EEPROM. Use this on a new install.
M501            ; Read back in the saved EEPROM.  

M190 S65        ; Not required, but having the printer at temperature helps accuracy
M104 S210       ; Not required, but having the printer at temperature helps accuracy

G28             ; Home XYZ.
G29 P1          ; Do automated probing of the bed.
G29 P2 B T      ; Manual probing of locations USUALLY NOT NEEDED!!!!
G29 P3 T        ; Repeat until all mesh points are filled in.

G29 T           ; View the Z compensation values.
G29 S1          ; Save UBL mesh points to EEPROM.
G29 F 10.0      ; Set Fade Height for correction at 10.0 mm.
G29 A           ; Activate the UBL System.
M500            ; Save current setup. WARNING: UBL will be active at power up, before any `G28`.
;---------------------------------------------
;--- Fine Tuning of the mesh happens below ---
;---------------------------------------------
G26 C P5.0 F3.0 ; Produce mesh validation pattern with primed nozzle (5mm) and filament diameter 3mm 
                ; PLA temperatures are assumed unless you specify, e.g., B 105 H 225 for ABS Plastic
G29 P4 T        ; Move nozzle to 'bad' areas and fine tune the values if needed
                ; Repeat G26 and G29 P4 T  commands as needed.

G29 S1          ; Save UBL mesh values to EEPROM.
M500            ; Resave UBL's state information.
```

**Comment by modem7** [comment]
I'm just surprised it worked in the older version! Could I have just been lucky and hit a bug that somehow got it to work with an inset of 10??? 

(Being 3am, it would be foolish for me to start UBL stuff now :D)

**Comment by modem7** [comment]
This is the result I get with mesh inset set to 0 (so same results, didn't wait to see yet if it tries to crash into the bed/z goes up to infinity)

[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/2087149/Marlin.zip)


![2018-06-09 17 26 30](https://user-images.githubusercontent.com/4349962/41193850-67cf5a68-6c0a-11e8-8a72-1a61f973d905.jpg)


  ```
//===========================================================================
  //========================= Unified Bed Leveling ============================
  //===========================================================================

  #define MESH_EDIT_GFX_OVERLAY     // Display a graphics overlay while editing the mesh

  #define MESH_INSET 0              // Mesh inset margin on print area
  #define GRID_MAX_POINTS_X 10      // Don't use more than 15 points per axis, implementation limited.
  #define GRID_MAX_POINTS_Y GRID_MAX_POINTS_X

  #define UBL_PROBE_PT_1_X 39       // Probing points for 3-Point leveling of the mesh
  #define UBL_PROBE_PT_1_Y 150
  #define UBL_PROBE_PT_2_X 39
  #define UBL_PROBE_PT_2_Y 20
  #define UBL_PROBE_PT_3_X 180
  #define UBL_PROBE_PT_3_Y 20

  #define UBL_MESH_EDIT_MOVES_Z     // Sophisticated users prefer no movement of nozzle
  #define UBL_SAVE_ACTIVE_ON_M500   // Save the currently active mesh in the current slot on M500

#elif ENABLED(MESH_BED_LEVELING)
```

Next step is using the cheat sheet instead of the UBL menu options

**Comment by modem7** [comment]
Same issue after clearing EEPROM as per the cheat sheet. Seems it's a bug of some sort.

Cleared EEPROM
Inset set to 0
Config and adv in previous post.

**Comment by thinkyhead** [comment]
Here are some things you can try:

- Check the accuracy of your probe using `M48`.
- Try subtracting 0.15 from your Z probe Z offset, then re-probe and re-test `G26`.
- Adjust the tension of your Y belt.

**Comment by modem7** [comment]
@thinkyhead If I go back to 1.1.4 everything works, the issue only presents on 1.1.8 latest bugfix. This isn't a hardware issue, it's software, either settings have changed on 1.1.8 which are conflicting, or it's an unusual bug I'm triggering.

Current bug:
1. Mesh isn't printing on whole bed
2. When printing mesh (seemingly always at the same point, near end of printing mesh), Z goes up or down to infinity (and crashing into bed/frame)

**Comment by Roxy-3D** [comment]
The circle drawing was changed to use G2 arcs a while ago...   Please turn off ARC_SUPPORT.   It is very likely G26 will start working again when you do that.    If that is the case, we will need to look closer at the Arc code and how that is implemented for UBL.   (The G2 / G3 arc code might not be calling the correct motion routine to schedule line segments.  Or, it may be the current[] and destination[] points are being left in the wrong state for G26.)

**Comment by modem7** [comment]
ARC_SUPPORT disabled but still not getting full bed being printed in this case. 

![2018-06-10 17 58 50](https://user-images.githubusercontent.com/4349962/41204127-62cc605a-6cd8-11e8-9423-76f69b897b22.jpg)

[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/2088055/Marlin.zip)

Got my settings as 

```
// G2/G3 Arc Support
//
//#define ARC_SUPPORT               // Disable this feature to save ~3226 bytes
#if ENABLED(ARC_SUPPORT)
  #define MM_PER_ARC_SEGMENT  1   // Length of each arc segment
  #define N_ARC_CORRECTION   25   // Number of intertpolated segments between corrections
  //#define ARC_P_CIRCLES         // Enable the 'P' parameter to specify complete circles
  //#define CNC_WORKSPACE_PLANES  // Allow G2/G3 to operate in XY, ZX, or YZ planes
```

```
// The size of the print bed
#define X_BED_SIZE 200
#define Y_BED_SIZE 200

// Travel limits (mm) after homing, corresponding to endstop positions.
#define X_MIN_POS 0
#define Y_MIN_POS 0
#define Z_MIN_POS 0
#define X_MAX_POS X_BED_SIZE
#define Y_MAX_POS Y_BED_SIZE
#define Z_MAX_POS 180
```

```
  #define UBL_PROBE_PT_1_X 39       // Probing points for 3-Point leveling of the mesh
  #define UBL_PROBE_PT_1_Y 150
  #define UBL_PROBE_PT_2_X 39
  #define UBL_PROBE_PT_2_Y 20
  #define UBL_PROBE_PT_3_X 180
  #define UBL_PROBE_PT_3_Y 20
```
```

#define X_PROBE_OFFSET_FROM_EXTRUDER 34  // X offset: -left  +right  [of the nozzle]
#define Y_PROBE_OFFSET_FROM_EXTRUDER -50 // Y offset: -front +behind [the nozzle]
#define Z_PROBE_OFFSET_FROM_EXTRUDER -1.19 // Z offset: -below +above  [the nozzle]
```

**Comment by Roxy-3D** [comment]
Yeah...   one of the developers is going to need to step through and debug things.    The problem is, I'm all setup for 32-bit work putting an RTOS under Marlin.      I might take a time out to do this...  but it would be better if somebody else figured out what is going wrong.

What ever is wrong, it isn't much.   It will probably be just a line or two of code to make everything right again.

**Comment by gloomyandy** [comment]
It might hekp if you could post a short video clip showing what is happening. That way anyone trying to reproduce the issue can be sure they are seeing the same problem. I've been following this, but I'm still not sure I understand exactly what the issue is.

**Comment by modem7** [comment]
One 30 sec video coming up! Apologies if it's sideways, apparently it doesn't enjoy going the proper way up.

https://www.dropbox.com/s/s1wwlmer8g30nnn/20180611_125424_1.mp4?dl=0

So as we can see, the mesh isn't being printed on the entire bed, and a lovely crunch as it goes down to infinity at the end, it certainly isn't off the mesh/bed at the time when it does it.

**Comment by gloomyandy** [comment]
Ouch! That's not nice! Could you provide a G29 T output for the mesh that was used when making the video? I assume it is different to the one above (I think you changed the MESH_INSET).

**Comment by gloomyandy** [comment]
Looking at the video up to the point of the nozzle crashing into the bed the test pattern seems to be complete and looks ok (perhaps a little squashed?). Are there problems that are not obvious in the video?

**Comment by modem7** [comment]
Aye, I think the Z offset may be a little off, haven't really dialled it in completely until I sort this out (constant reflashing firmware n all).

But apart from the two UBL issues above, haven't noticed anything further. But it certainly looks like it's not seeing a 200x200 bed, but something smaller. Which is odd because the probe homes exactly as intended (200x200/2), and it's probing (as much as it can) the full bed.

```
15:03:24.325 : Bed Topography Report:
15:03:24.356 : (0,9)                                                                   (9,9)
15:03:24.372 : (34,150)                                                                (195,150)
15:03:24.606 : 0.678    0.630  [ 0.367]   0.167    0.058   -0.154   -0.269   -0.510   -0.831   -1.065
15:03:24.606 : 
15:03:24.778 : 0.680    0.503    0.436    0.165    0.053    0.000   -0.388   -0.582   -0.724   -1.123
15:03:24.778 : 
15:03:24.980 : 0.740    0.439    0.404    0.292    0.005   -0.219   -0.473   -0.677   -0.814   -1.200
15:03:24.980 : 
15:03:25.136 : -0.211   -0.458   -0.702   -0.956   -1.215
15:03:25.136 : 
15:03:25.292 : 0.690    0.479    0.289    0.167   -0.077   -0.269   -0.480   -0.719   -0.941   -1.205
15:03:25.292 : 
15:03:25.433 : 0.675    0.374    0.429    0.165   -0.037   -0.266   -0.475   -0.707   -0.839   -1.195
15:03:25.433 : 
15:03:25.589 : 0.765    0.531    0.284    0.212   -0.069   -0.274   -0.485   -0.635   -0.871   -1.235
15:03:25.589 : 
15:03:25.745 : 0.780    0.451    0.292    0.212    0.035   -0.261   -0.480   -0.667   -0.943   -1.215
15:03:25.745 : 
15:03:25.901 : 0.748    0.523    0.526    0.197   -0.032   -0.156   -0.470   -0.595   -0.916   -1.252
15:03:25.901 : 
15:03:26.119 : -0.505   -0.759   -0.946   -1.252
15:03:26.150 : (34,5)                                                                    (195,5)
15:03:26.166 : (0,0)                                                                     (9,0)

```

**Comment by gloomyandy** [comment]
Are there some values missing from that G29 T? There is something odd going on if not. Look at the 4th and 10th line from the top they only have 4 values instead of 10. It also seems a little odd that your mesh starts at 34, 5 and runs to 195, 150, your bed is supposed to be 200x200. Can you post and up to date copy of your configuration files. Make sure you include both.

**Comment by modem7** [comment]
I didn't do any smart fill in for the post (unlike the previous ones), just in case something like this came up and it was obscured. 

[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/2090773/Marlin.zip)

Could it be related to the settings in the above coment? (https://github.com/MarlinFirmware/Marlin/issues/10963#issuecomment-396064433)

_I've also edited my previous comment so that it's a direct copy instead of tidied up, just in case it screwed with formatting!_

**Comment by Roxy-3D** [comment]
You should have your mesh fully populated with values.

**Comment by modem7** [comment]
Normally I run the G29 under the step by step menu (result above), then smart fill in a couple of times, and only then do I print the mesh (the video was after smart fill in). 

Would you like me to post the values after smart fill in (It'll be similar to the one above in https://github.com/MarlinFirmware/Marlin/issues/10963#issuecomment-395889488)

**Comment by modem7** [comment]
Just G29
```
18:16:22.986 : Bed Topography Report:
18:16:23.017 : (0,9)                                                                   (9,9)
18:16:23.048 : (34,150)                                                                (195,150)
18:16:23.251 : 0.630    0.586  [ 0.354]   0.187    0.070   -0.124   -0.336   -0.485   -0.854   -1.100
18:16:23.251 : 
18:16:23.407 : 0.700    0.523    0.426    0.117    0.058   -0.007   -0.368   -0.595   -0.707   -1.113
18:16:23.407 : 
18:16:23.563 : 0.698    0.439    0.357    0.239   -0.039   -0.234   -0.480   -0.682   -0.836   -1.220
18:16:23.563 : 
18:16:23.781 : -0.069   -0.241   -0.475   -0.722   -0.919   -1.202
18:16:23.781 : 
18:16:23.937 : 0.695    0.474    0.309    0.152   -0.049   -0.288   -0.530   -0.702   -0.921   -1.255
18:16:23.937 : 
18:16:24.093 : 0.653    0.424    0.389    0.150   -0.124   -0.353   -0.490   -0.719   -0.874   -1.235
18:16:24.093 : 
18:16:24.249 : 0.718    0.511    0.344    0.167   -0.022   -0.251   -0.518   -0.642   -0.963   -1.217
18:16:24.249 : 
18:16:24.405 : 0.785    0.469    0.272    0.192    0.115   -0.241   -0.450   -0.727   -0.936   -1.252
18:16:24.405 : 
18:16:24.561 : 0.743    0.474    0.551    0.272   -0.015   -0.234   -0.488   -0.625   -0.966   -1.272
18:16:24.561 : 
18:16:24.717 : -0.318   -0.520   -0.767   -1.001   -1.245
18:16:24.749 : (34,5)                                                                    (195,5)
18:16:24.764 : (0,0)                                                                     (9,0)
```
After smart fill-in
```
18:16:51.159 : echo:enqueueing "G29 P3 T0"
18:16:51.175 : Bed Topography Report:
18:16:51.206 : (0,9)                                                                   (9,9)
18:16:51.222 : (34,150)                                                                (195,150)
18:16:51.456 : 0.630    0.586  [ 0.354]   0.187    0.070   -0.124   -0.336   -0.485   -0.854   -1.100
18:16:51.456 : 
18:16:51.643 : 
18:16:51.643 : 
18:16:51.799 : 0.698    0.439    0.357    0.239   -0.039   -0.234   -0.480   -0.682   -0.836   -1.220
18:16:51.799 : 
18:16:51.955 : 0.660    0.486    0.352    0.192   -0.069   -0.241   -0.475   -0.722   -0.919   -1.202
18:16:51.955 : 
18:16:52.111 : 0.695    0.474    0.309    0.152   -0.049   -0.288   -0.530   -0.702   -0.921   -1.255
18:16:52.111 : 
18:16:52.267 : 0.653    0.424    0.389    0.150   -0.124   -0.353   -0.490   -0.719   -0.874   -1.235
18:16:52.267 : 
18:16:52.407 : 0.718    0.511    0.344    0.167   -0.022   -0.251   -0.518   -0.642   -0.963   -1.217
18:16:52.407 : 
18:16:52.563 : 0.785    0.469    0.272    0.192    0.115   -0.241   -0.450   -0.727   -0.936   -1.252
18:16:52.563 : 
18:16:52.719 : -0.234   -0.488   -0.625   -0.966   -1.272
18:16:52.719 : 
18:16:52.875 : 0.628    0.431    0.314    0.155   -0.099   -0.318   -0.520   -0.767   -1.001   -1.245
18:16:52.907 : (34,5)                                                                    (195,5)
18:16:52.922 : (0,0)                                                                     (9,0)

```


Also, just had a thought, which was silly that I didn't notice it before. 

It's currently only printing the mesh where the probe can reach, whereas the smart fill should be filling in the points where the probe can't reach. This was handled in 1.1.4, but in 1.1.8 it seems to be totally unhandled (see images below):

![2018-06-11 18 18 27](https://user-images.githubusercontent.com/4349962/41246969-84c5f726-6da4-11e8-843b-f67eb1794ecb.jpg)
![2018-06-11 18 18 16](https://user-images.githubusercontent.com/4349962/41246970-84e96454-6da4-11e8-89ee-2cef23c89b24.jpg)
![2018-06-11 18 18 12](https://user-images.githubusercontent.com/4349962/41246972-8514cbd0-6da4-11e8-873e-f0454358092c.jpg)

**Comment by gloomyandy** [comment]
Yes please post the mesh as it is when you actually use it. I can't see any reason from your configuration why the mesh would start at (34,5) or end at (195, 150). I did notice that you have a MESH_INSET of 5 set. Personally I think that would be better as 0 so that the mesh covers all of the printable area.

**Comment by modem7** [comment]
@gloomyandy all done for you sire: https://github.com/MarlinFirmware/Marlin/issues/10963#issuecomment-396319655

The second entry (after smart fill in) is what I'd use normally.

I originally (in 1.1.4) had mesh inset as 10, tried 0 during the course of this issue, and was trying 5 just in case something positive happened (guess I didn't change it back!)

**Comment by gloomyandy** [comment]
There is something very odd going on with those posted mesh listings. Do they actually look like that or is there some problem with the way you are posting them? There seem to be values missing (in the case of the second one entire lines, you only seem to have 9 rather than 10!). Perhaps it is some problem with my browser, do you see a full mesh posted above?

**Comment by modem7** [comment]
Just in case there are, here is the notepad file, directly copied from repetier:

[ubl.txt](https://github.com/MarlinFirmware/Marlin/files/2091099/ubl.txt)

**Comment by gloomyandy** [comment]
I've just noticed that your X and Y probe offsets are 34 and -50 which is exactly the same as the odd offsets in your mesh (starts at X = 34 and has a Y size that is 50 smaller then the bed). I have never seen UBL construct a mesh this way. @Roxy-3D  isn't a UBL mesh normally the same size as the bed minus any MESH_INSET?

**Comment by gloomyandy** [comment]
Yep still has missing data, could you try G29 T1 that may have a more compact format.

**Comment by modem7** [comment]
Since I started using the BLTouch, I've used this mount + their settings:

https://www.thingiverse.com/thing:2394428

> My Marlin firmware settings for this are as follows:
> 
> These settings under the Z probe options:
> 
> define X_PROBE_OFFSET_FROM_EXTRUDER 34 // X offset: -left +right [of the nozzle]
> define Y_PROBE_OFFSET_FROM_EXTRUDER -50 // Y offset: -front +behind [the nozzle]
> define Z_PROBE_OFFSET_FROM_EXTRUDER -1.8 // Z offset: -below +above [the nozzle]
> Auto bed leveling (Bi-linear):
> 
> // Set the boundaries for probing (where the probe can reach).
> 
> define LEFT_PROBE_BED_POSITION 35
> define RIGHT_PROBE_BED_POSITION 194
> define FRONT_PROBE_BED_POSITION 10
> define BACK_PROBE_BED_POSITION 150

(Will do a T1 now)

After Smart Fill:
``` 
19:00:21.527 : Bed Topography Report for CSV:
19:00:21.761 : 0.591,0.655,0.362,0.232,-0.039,-0.194,-0.318,-0.562,-0.881,-1.095
19:00:21.902 : 0.668,0.521,0.401,0.207,0.023,-0.107,-0.351,-0.605,-0.754,-1.130
19:00:22.058 : 0.556,0.436,0.324,0.212,-0.037,-0.283,-0.443,-0.719,-0.909,-1.222
19:00:22.214 : ,-0.234,-0.445,-0.704,-0.919,-1.222
19:00:22.370 : 0.685,0.484,0.322,0.132,-0.114,-0.274,-0.513,-0.704,-0.869,-1.225
19:00:22.526 : 0.576,0.444,0.399,0.127,-0.039,-0.236,-0.520,-0.722,-0.894,-1.210
19:00:22.682 : 0.745,0.451,0.282,0.115,-0.039,-0.313,-0.515,-0.635,-0.986,-1.217
19:00:22.838 : 0.725,0.479,0.294,0.197,-0.034,-0.281,-0.510,-0.739,-0.884,-1.242
19:00:23.056 : 0.693,0.481,0.436,0.207,-0.015,-0.196,-0.443,-0.557,-0.981,-1.237
19:00:23.212 : ,-0.503,-0.819,-1.033,-1.257
```

**Comment by gloomyandy** [comment]
That still seems to have data missing from it. I wonder if your system is dropping characters for large output from Marlin.

I'm at a lost as to why your mesh has those start and end points. Are you sure the configuration files you have posted are what is actually being used to compile your firmware (the mesh looks more like the result of using MESH_BED_LEVELLING not UBL. Have you made any modifications to any of the other files from the download? I notice that your configutation.h file contains a version number 010107 the current bugfix is 010109. You may be better off starting again with your config file to make sure that everything is up to date. 

Another thought, are you sure the firmware on your machine is the bugfix-1.1.x version? Older versions of Marlin used to configure the mesh as you are seeing it when using UBL.

**Comment by thinkyhead** [comment]
Whenever there are homing or leveling issues, we now ask everyone to follow a standard procedure to gather more information so that we're not just taking stabs in the dark. Here is the boilerplate:

- Download Marlin `bugfix-1.1.x` or `bugfix-2.0.x` to test with the latest code.
- Enable `DEBUG_LEVELING_FEATURE` and re-flash the firmware.
- Connect to your printer from host software such as Cura, Printrun or Repetier Host.
- Issue the command `M111 S247` to enable maximum logging.
- Perform a `G28` to do your standard homing procedure.
- Do a `G29` to probe the bed. This will also enable bed leveling.
- Copy the log output into a **.TXT file** and attach it to your next reply.

**Comment by modem7** [comment]
OK, I've redownloaded the latest bugfix, redid the config and adv from scratch and enabled debug as well.

Also tested the mesh just in case, and unfortunately still not printing entire bed (and I'm *assuming* that it'll probably crash down onto the bed as well, don't want to repeat that TOO much times for obvious reasons!)

[Marlin.zip](https://github.com/MarlinFirmware/Marlin/files/2091765/Marlin.zip)

**Comment by gloomyandy** [comment]
The two configuration files you have uploaded are not from the latest bugfix release (the configuration.h file still shows the version as 010107, the latest bugfix is 010109). Did you attach the correct files? Also the log file seems to indicate that the firmware used to generate the log was version 1.1.8 not the bugfix version. What did you do to download/configure/build/install the latest bugfix?

**Comment by modem7** [comment]
Hmmmm - I sincerely apologise. I will see what the heck is going on my end, because things should be uploaded as intended. I will re-do everything tomorrow, double checking things are 010109.

Thank you so far guys, sorry for the idiocy on this end!

**Comment by Roxy-3D** [comment]
The bugfix version of UBL is more stable than the release version....   It would be nice to get the release version caught up.

**Comment by lrpirlet** [comment]
@modem7 @Gloom

**Comment by lrpirlet** [comment]
@modem7 @gloomyandy
You have AUTO_REPORT_TEMPERATURES in Configuration_adv.h enabled…. Please issue M155 S20 to stop reporting temperature long enough for G29 T to complete…
If you have Repetier, you may want to issue M155 S1 after…

Sequence becomes
```
M155 S25
G29 T
M155 S1
```
Ref: issue #7021

**Comment by modem7** [comment]
OK, just uploading these as a matter of course.

As usual, it looks like human stupidity and a pursuit of shortcuts *may* have been the cause here. 

Give me 30 minutes to see if everything is working and I'll confirm the process I used and how I was a moron so that if further muppets come along, we can make sure they're not being as silly as I am (just need to verify to see if head crashes or rises, then we'll go on from there)

[debug.txt](https://github.com/MarlinFirmware/Marlin/files/2095332/debug.txt)
[bed levelling.txt](https://github.com/MarlinFirmware/Marlin/files/2095333/bed.levelling.txt)

**Comment by modem7** [comment]
Confirmed. ID10T error.

I was using https://github.com/akaJes/marlin-config to setup and transfer my configs over. 

This obviously didn't do a very good job, and messed a few things up, I suspect that it was trying to mix 1.1.8 with bugfix files which really messed things up. 

Once I downloaded the bugfix + modified the files manually, everything seems to work as expected. 

Apologies all who helped me, and thank you so much for spending your time to do so! You've been an immense help. Sorry for the idiocy!
